pennylane/tape/tapes/reversible.py

```diff
@@ -180,26 +192,71 @@ def analytic_pd(self, idx, device, params=None, **options):
         else:
             generator, multiplier = op.generator
 
-        generator = generator(wires)
+        # construct circuit to compute differentiated state
+        between_ops_inverse = [copy.deepcopy(op) for op in between_ops[::-1]]

```

Comment:
Another deep copy hidden here. I wonder if it is truly necessary (either right now, or longer-term)?

Reply:
Was thinking about it a bit more. 

Could it make sense to initially just make one single list of `between_ops_inverse` which contains all ops between the final measurement and the first parameterized gate, and store this somewhere in the class? 

Then the other various tapes could be made by just subsetting that largest case (wouldn't need to mutate any parameters!), and potentially adding a small number of extra ops as needed?

Reply:
@co9olguy this actually shouldn't be a deepcopy, I've reverted it to simply `copy(op)`, as it was before! Thanks for catching this.